src/adsbrain_server.h

```diff
@@ -0,0 +1,113 @@
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <mutex>
+
+#include "common.h"
+#include "dirent.h"
+#include "http_server.h"
+#include "triton/core/tritonserver.h"
+
+namespace triton { namespace server {
+
+// Handle AdsBrain HTTP requests to inference server APIs
+class AdsBrainAPIServer : public HTTPAPIServer {
+  enum RequestType { TRITON, ADSBRAIN_BOND };
+
+  public:
+  static TRITONSERVER_Error* Create(
+      const std::shared_ptr<TRITONSERVER_Server>& server,
+      triton::server::TraceManager* trace_manager,
+      const std::shared_ptr<SharedMemoryManager>& smb_manager,
+      const int32_t port, const std::string address, const int thread_cnt,
+      std::unique_ptr<HTTPServer>* http_server);
+
+  class AdsBainInferRequestClass : public HTTPAPIServer::InferRequestClass {
+   public:
+    explicit AdsBainInferRequestClass(
+        TRITONSERVER_Server* server, evhtp_request_t* req,
+        DataCompressor::Type response_compression_type, RequestType request_type)
+        : InferRequestClass(server, req, response_compression_type),
+        request_type_(request_type)
+    {
+    }
+
+    TRITONSERVER_Error* FinalizeResponse(
+        TRITONSERVER_InferenceResponse* response) override;
+
+   private:
+    TRITONSERVER_Error* FinalizeResponseInTritonFormat(
+        TRITONSERVER_InferenceResponse* response)
+    {
+      return InferRequestClass::FinalizeResponse(response);
+    }
+
+    TRITONSERVER_Error* FinalizeResponseInBondFormat(
+        TRITONSERVER_InferenceResponse* response);
+
+    RequestType request_type_ = RequestType::ADSBRAIN_BOND;
+  };
+
+
+ protected:
+  std::unique_ptr<InferRequestClass> CreateInferRequest(
+      evhtp_request_t* req) override
+  {
+    return std::unique_ptr<InferRequestClass>(new AdsBainInferRequestClass(
+        server_.get(), req, GetResponseCompressionType(req), request_type_));
+  }
+
+ private:
+  explicit AdsBrainAPIServer(
+      const std::shared_ptr<TRITONSERVER_Server>& server,
+      triton::server::TraceManager* trace_manager,
+      const std::shared_ptr<SharedMemoryManager>& shm_manager,
+      const int32_t port, const std::string address, const int thread_cnt)
+      : HTTPAPIServer(
+            server, trace_manager, shm_manager, port, address, thread_cnt)
+  {
+    auto request_type =
+          GetEnvironmentVariableOrDefault("AB_REQUEST_TYPE", "ADSBRAIN_BOND");
+      LOG_INFO << "[adsbrain] AB_REQUEST_TYPE=" << request_type;
+      if (request_type == "TRITON") {
+        request_type_ = RequestType::TRITON;
+      } else if (request_type == "ADSBRAIN_BOND") {
+        request_type_ = RequestType::ADSBRAIN_BOND;
+      } else {
+        LOG_INFO << request_type
+                 << " is not supported, so use the default triton format."
```

Comment:
nit: `ADSBRAIN_BOND` is the default version, so update the log info to be `so use the default ADSBRAIN_BOND format.`